ERROR while running task C:\Users\xxxxxx\GitHub\xxxx-app:serve with message'spawn vue-cli-service ENOENT'

### Version
3.0.0-rc.10

### Node and OS info
Windows 10 Enterprise / Node 8.9.1 / Yarn 1.7.0 / GitBash (MINGW64)

### Steps to reproduce
1. Run default "serve" task
2. Stop default "serve" task

### What is expected?
A clean shutdown of the task without error

### What is actually happening?
ERROR while running task C:\Users\xxxx\GitHub\xxxx-app:serve with message'spawn vue-cli-service ENOENT'

{ Error: spawn vue-cli-service ENOENT
at notFoundError (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\cross-spawn\lib\enoent.js:6:26)
at verifyENOENT (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\cross-spawn\lib\enoent.js:40:16)
at ChildProcess.cp.emit (C:\Users\xxxx\AppData\Local\Yarn\Data\global\node_modules\cross-spawn\lib\enoent.js:27:25)
at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
code: 'ENOENT',
errno: 'ENOENT',
syscall: 'spawn vue-cli-service',
path: 'vue-cli-service',
spawnargs: [ 'serve', '--mode', 'development', '--dashboard' ] }

<!-- generated by vue-issues. DO NOT REMOVE -->

Vue CLI v3.0.0-rc.10
✨  Creating project in /Users/raymond/stack/sites/xml.
🗃  Initializing git repository...
⚙  Installing CLI plugins. This might take a while...

npm ERR! code ETARGET
npm ERR! notarget No matching version found for @vue/cli-plugin-babel@^3.0.0-rc.11
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'xml'
npm ERR! notarget 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/raymond/.npm/_logs/2018-08-07T15_32_34_730Z-debug.log
 ERROR  command failed: npm install --loglevel error

--- a/packages/@vue/cli-ui/apollo-server/connectors/tasks.js
+++ b/packages/@vue/cli-ui/apollo-server/connectors/tasks.js
@@ -17,6 +17,7 @@ const { notify } = require('../util/notification')
 
 const MAX_LOGS = 2000
 const VIEW_ID = 'vue-project-tasks'
+const WIN_ENOENT_THRESHOLD = 500 // ms
 
 const tasks = new Map()
 
@@ -411,6 +412,11 @@ async function run (id, context) {
     child.on('exit', onExit)
 
     child.on('error', error => {
+      const duration = Date.now() - task.time
+      // hackish workaround for https://github.com/vuejs/vue-cli/issues/2096
+      if (process.platform === 'win32' && error.code === 'ENOENT' && duration > WIN_ENOENT_THRESHOLD) {
+        return onExit(null)
+      }
       updateOne({
         id: task.id,
         status: 'error'

